catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -422,23 +433,38 @@ private ContentItem readContent(URI uri) throws StorageException {
     }
 
     String mimeType = DEFAULT_MIME_TYPE;
-    long size = 0;
-    ByteSource byteSource;
+    byte[] encryptedBytes;
+    byte[] plainBytes;
+    ByteSource plainByteSource;
 
     try (InputStream fileInputStream =
         file != null ? Files.newInputStream(file) : reference.toURL().openStream()) {
-      mimeType = mimeTypeMapper.guessMimeType(fileInputStream, extension);
+      encryptedBytes = ByteStreams.toByteArray(fileInputStream);
+      plainBytes = encryptorDecryptor.decrypt(encryptedBytes);
```

Comment:
❗️ We can't pass the bytes around through memory like this. We'll need to pass streams around.